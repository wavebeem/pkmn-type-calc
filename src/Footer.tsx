import * as React from "react";

import { ReleaseNotes } from "./ReleaseNotes";

export function Footer() {
  const [isReleaseNotesOpen, setIsReleaseNotesOpen] = React.useState(false);
  const year = new Date().getFullYear();
  return (
    <footer className="black dark-gray bt b--black-05 ph3 pv2">
      <div className="mw7 center">
        <p>
          <a
            href="#"
            className="ThickUnderline b near-black hover-gray DashedFocus"
            onClick={(event) => {
              event.preventDefault();
              setIsReleaseNotesOpen(!isReleaseNotesOpen);
            }}
          >
            {isReleaseNotesOpen ? (
              <React.Fragment>Hide release notes</React.Fragment>
            ) : (
              <React.Fragment>Show release notes</React.Fragment>
            )}
          </a>
        </p>
        {isReleaseNotesOpen ? <ReleaseNotes /> : null}
        <p>
          Pokémon &copy; 2002-{year} Pokémon. &copy; 1995-{year}{" "}
          Nintendo/Creatures Inc./GAME FREAK inc. &trade;, &reg; and Pokémon
          character names are trademarks of Nintendo.
        </p>
        <p>
          No copyright or trademark infringement is intended in using Pokémon
          content on this page.
        </p>
        <p>
          pkmn.help &copy; 2013-{year}{" "}
          <a
            href="https://mockbrian.com"
            className="b ThickUnderline near-black hover-gray DashedFocus"
          >
            Brian Mock
          </a>
          .
        </p>
      </div>
    </footer>
  );
}
