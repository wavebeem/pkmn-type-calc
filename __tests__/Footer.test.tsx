import React from "react";
import { ShallowWrapper, shallow } from "enzyme";

import Footer from "../src/Footer";

describe("Footer", () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Footer />);
  });

  it("renders correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
