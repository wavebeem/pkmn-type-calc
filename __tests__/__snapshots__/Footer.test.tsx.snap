// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="black dark-gray bt b--black-05 ph3 pv2"
>
  <div
    className="mw7 center"
  >
    <p>
      <a
        className="ThickUnderline b near-black hover-gray DashedFocus"
        href="#"
        onClick={[Function]}
      >
        Show release notes
      </a>
    </p>
    <p>
      Pokémon © 2002-
      2020
       Pokémon. © 1995-
      2020
       
      Nintendo/Creatures Inc./GAME FREAK inc. ™, ® and Pokémon character names are trademarks of Nintendo.
    </p>
    <p>
      No copyright or trademark infringement is intended in using Pokémon content on this page.
    </p>
    <p>
      pkmn.help © 2013-
      2020
       
      <a
        className="b ThickUnderline near-black hover-gray DashedFocus"
        href="https://mockbrian.com"
      >
        Brian Mock
      </a>
      .
    </p>
  </div>
</footer>
`;
